### ISSUE TITLE
Hard Wired Numpy Header files in setup.py install

### ISSUE THREAD

**Issue by janithwanni** [clips/metameric#2]
There seems to be an error when trying to run setup.py install within an virtual environment where the script gives out the following error.
`creating build/temp.macosx-10.6-intel-3.5/metameric/core`

`/usr/bin/clang -fno-strict-aliasing -Wsign-compare -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -arch i386 -arch x86_64 -g -I/Library/Frameworks/Python.framework/Versions/3.5/include/python3.5m -c metameric/core/metric.c -o build/temp.macosx-10.6-intel-3.5/metameric/core/metric.o`

`metameric/core/metric.c:444:10: fatal error: 'numpy/arrayobject.h' file not found`

`#include "numpy/arrayobject.h"`
`                 ^~~~~~~~~~~~~~~~~~~~~`
`1 error generated.
error: command '/usr/bin/clang' failed with exit status 1`

I believe the issue can be remedied by editing the setup.py file to include the numpy folder as mentioned in this answer [here](https://stackoverflow.com/a/35784219/1941842)
I will write the changes and see if they are working and submit a pull request @stephantul ?

**Comment by stephantul** [comment]
Hey @janithwanni, 

Thanks for reporting the issue. I have not encountered this error before, will try to reproduce. I do think the `setup.py` file includes the fix proposed in the link you posted (i.e., we do include the headers using `np.getinclude()` (see [here](https://github.com/clips/metameric/blob/57daf05c8c49e3109de1a7963b7aa72b19e8a157/setup.py#L18))). Can you try to run the setup again using `--cython` as an extra argument? I assumed that the included `metric.c`, which is precompiled, would work for users, but I guess in this case it doesn't. If adding `--cython` does the trick, I will rewrite the setup to do this by default.

As an aside, does this only happen within the virtualenv, or do you also encounter it outside of the environment? In any case, if you do find a fix, a PR is always welcome. 

Thanks again,
Stéphan

**Comment by janithwanni** [comment]
So far my observations were this, 
* Adding the `--cython` argument didn't make a difference inside virtualenv
* Running the setup.py install with and without `--cython` resulted in the same error (starting to think that this might just be an OSX error with Clang)
* Implementing what I proposed did fix the issue but when trying to run metameric.web module, matplotlib started complaining about the use of a regular build of python in virtualenv. (From this [link](https://matplotlib.org/faq/osx_framework.html ): In a virtualenv, a non-framework build is used even when the environment is created from a framework build (virtualenv bug #54, virtualenv bug #609).

I inspected the code of setup.py a bit more and read the [documentation](http://cython.readthedocs.io/en/latest/src/reference/compilation.html) which was written for distutils while the setup.py had setuptools. After reading the code of the setuptools extension.py file in their repo, it was clear that they provided a simple wrapper for distutils extension class. (simply putting down the path of my search here for future references 😄 ) 
So as the documentation describes I modified the line you mentioned into this way
```
extensions = cythonize([Extension("metameric.core.metric",
                                      ["metameric/core/metric.pyx"],
                                      include_dirs=[np.get_include()] ) 
                                      ],
                           include_path=[np.get_include()])
```
the install worked but then im still running into the following error while trying to get the metameric.web module up and running. (the error is the same for the example.csv attempt)
```
File "/Users/janith/Documents/Python_Projects/nlp-playground/metameric/metameric/core/layer.py", line 3, in <module>
    from .metric import strength
ImportError: No module named 'metameric.core.metric'
``` 

I'm starting to think this might be an OSX error so I will try testing it on a Windows machine or a Linux VM
